src/resolvers/block.ts

```diff
@@ -36,22 +38,66 @@ const blocks: IFieldResolver<any, any> = (obj, { numbers, hashes }: IBlocksArgs)
   if (numbers && hashes) {
     throw new Error('Only one of blocks or hashes should be provided.');
   }
-
   const f = v => web3.eth.getBlock(v, true);
   return Promise.all(numbers ? numbers.map(f) : hashes.map(f));
 };
 
+// Select multiple blocks.
+interface IBlocksRangeArgs {
+  numbers?: number[];
+  hashes?: string[];
+}
+
+const blocksRange: IFieldResolver<any, any> = async (obj, { numbers, hashes }: IBlocksRangeArgs) => {
+  if (numbers && hashes) {
+    throw new Error('Only one of blocks or hashes should be provided.');
+  }
+  type funcType = (hash: string, callback: (err: Error, res: BlockWithoutTransactionData) => void) => void;
+  
+  const getBlock: ((string) => Promise<BlockWithoutTransactionData>) = (hash) => {
+    return new Promise((res, rej) => {
+      web3.eth.getBlock(hash, (err, block) => {
+        if (err) {
+          rej(err);
+        } else {
+          res(block);
+        }
+      });
+    });
+  };
+
+  let start: number, end:number;
+
+  if (numbers && numbers.length == 2) {
+    [start, end] = numbers;
+  } else if (hashes && hashes.length == 2) {
+    const startBlock = await getBlock(hashes[0]);
+    start = startBlock.number;
+    const endBlock = await getBlock(hashes[1]);
+    end = endBlock.number;
+  } else {
+    throw new Error('Only the start and end should be provided.');
+  }
+
+  const f = v => web3.eth.getBlock(v, true);
+
+  const blocksRange = Array.from({length: (end - start + 1)}, (v, k) => k + start);
+
+  return Promise.all(blocksRange.map(f));
+};
+
 const transactions = (obj, args) =>
   obj.transactions.map(tx => ({ ...tx, from: { address: tx.from }, to: { address: tx.to }, inputData: tx.input }));
 
 const resolvers: IResolvers = {
   Query: {
     block,
     blocks,
+    blocksRange,
   },
   Block: {
     transactions,
   },
 };
 
-export default resolvers;
+export default resolvers;
```

Comment:
Are these line breaks being added by some tool? Are they added by the linter on commit? Or by your formatting configuration in VS Code? These changes muddy up the diff a little bit, so it's a good idea to review your diff before making the commit, and reverting such changes if possible.

Reply:
Strange. Not sure what's causing it but will double check in the future.